### PR TITLE
Add unit tests for null cpu for null parameters for set allocated cpu

### PR DESCRIPTION
**Pull request by charlesweng**
### Description

This test case tests the various if branches inside CpuAllocation that can result in null pointer exceptions. It is important to ensure that all arguments inside the cpu test within a host are allocated correctly, otherwise processes and tasks should not be runnable on the cloud system.

### Types of changes

- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Enhancement (improves an existing feature and functionality)
- [x] Cleanup (Code refactoring and cleanup, that may add test cases)
- [ ] build/CI

#### Feature/Enhancement Scale

- [ ] Major
- [x] Minor

#### Bug Severity

- [ ] BLOCKER
- [ ] Critical
- [ ] Major
- [x] Minor
- [ ] Trivial

### How Has This Been Tested?

This has been tested with the standard maven install lifecycle that includes all the tests

### CODE REVIEW

File: plugins/metrics/src/test/java/org/apache/cloudstack/response/HostMetricsResponseTest.java [hunk 1]
```diff
@@ -56,4 +56,15 @@ public void testSetCpuAllocatedWithValidCpu() {
...
+    public void testSetCpuAllocatedWithNullCpu() {
+        String expected = null;
+        final HostMetricsResponse hostResponse = new HostMetricsResponse();
+        hostResponse.setCpuAllocated(null, 10, 1000L);
+        Assert.assertEquals(expected, hostResponse.getCpuAllocatedGhz());
+        hostResponse.setCpuAllocated(String.format("50%s25%%", decimalSeparator), null, 1000L);
+        Assert.assertEquals(expected, hostResponse.getCpuAllocatedGhz());
+        hostResponse.setCpuAllocated(String.format("50%s25%%", decimalSeparator), 10, null);
+        Assert.assertEquals(expected, hostResponse.getCpuAllocatedGhz());
+    }
```
Comment (JoaoJandre) [inline plugins/metrics/src/test/java/org/apache/cloudstack/response/HostMetricsResponseTest.java:69]: Each unit test should test only one test case. Please break this into different tests.

**Comment by charlesweng** [comment]
@JoaoJandre split up the tests, thank you for the feedback!

**Review by DaanHoogland — APPROVED** [review]
clgtm

**Review by JoaoJandre — APPROVED** [review]
CLGTM
